cmd/controller/controller.go

```diff
@@ -225,12 +225,23 @@ func (c *CmdOpts) startController() error {
 	// common factory to get the admin kube client that's needed in many components
 	adminClientFactory := kubernetes.NewAdminClientFactory(c.K0sVars)
 
+	enableKonnectivity := true
+	if c.SingleNode {
+		enableKonnectivity = false
+	} else if stringslice.Contains(c.DisableComponents, constant.KonnectivityServerComponentName) {
+		if c.EnableWorker {
+			enableKonnectivity = false
```

Comment:
I think we should still just disable konnectivity is user has given the flag to disable it. There are some use cases where users want to run only the kube API and nothing else, in this case there's never a need for konnectivity at all.

Reply:
OK. Let's respect user's instructions.